airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/NoOpTyperDeduperWithV1V2Migrations.java

```diff
@@ -55,38 +54,32 @@ public NoOpTyperDeduperWithV1V2Migrations(final SqlGenerator<DialectTableDefinit
         new BasicThreadFactory.Builder().namingPattern(TYPE_AND_DEDUPE_THREAD_NAME).build());
   }
 
-  private void prepareSchemas(final ParsedCatalog parsedCatalog) throws Exception {
-    final var rawSchema = parsedCatalog.streams().stream().map(stream -> stream.id().rawNamespace());
-    final var finalSchema = parsedCatalog.streams().stream().map(stream -> stream.id().finalNamespace());
-    final var createAllSchemasSql = Streams.concat(rawSchema, finalSchema)
-        .filter(Objects::nonNull)
-        .distinct()
-        .map(sqlGenerator::createSchema)
-        .toList();
-    destinationHandler.execute(Sql.concat(createAllSchemasSql));
-  }
-
   @Override
   public void prepareTables() throws Exception {
-    log.info("ensuring schemas exist for prepareTables with V1V2 migrations");
-    prepareSchemas(parsedCatalog);
-    final Set<CompletableFuture<Optional<Exception>>> prepareTablesTasks = new HashSet<>();
-    for (final StreamConfig stream : parsedCatalog.streams()) {
-      prepareTablesTasks.add(CompletableFuture.supplyAsync(() -> {
-        // Migrate the Raw Tables if this is the first v2 sync after a v1 sync
-        try {
-          log.info("Migrating V1->V2 for stream {}", stream.id());
-          v1V2Migrator.migrateIfNecessary(sqlGenerator, destinationHandler, stream);
-          log.info("Migrating V2 legacy for stream {}", stream.id());
-          v2TableMigrator.migrateIfNecessary(stream);
-          return Optional.empty();
-        } catch (final Exception e) {
-          return Optional.of(e);
-        }
-      }, executorService));
+    try {
+      log.info("ensuring schemas exist for prepareTables with V1V2 migrations");
+      prepareAllSchemas(parsedCatalog, sqlGenerator, destinationHandler);
+      final Set<CompletableFuture<Optional<Exception>>> prepareTablesTasks = new HashSet<>();
+      for (final StreamConfig stream : parsedCatalog.streams()) {
+        prepareTablesTasks.add(CompletableFuture.supplyAsync(() -> {
+          // Migrate the Raw Tables if this is the first v2 sync after a v1 sync
+          try {
+            log.info("Migrating V1->V2 for stream {}", stream.id());
+            v1V2Migrator.migrateIfNecessary(sqlGenerator, destinationHandler, stream);
+            log.info("Migrating V2 legacy for stream {}", stream.id());
+            v2TableMigrator.migrateIfNecessary(stream);
+            return Optional.empty();
+          } catch (final Exception e) {
+            return Optional.of(e);
+          }
+        }, executorService));
+      }
+      CompletableFuture.allOf(prepareTablesTasks.toArray(CompletableFuture[]::new)).join();
```

Comment:
nit: there's nothing wrong with this, but consider using the java.util.Streams utils:
```
var prepareTablesTasks = parsedCatalog.streams().stream()
    .map(stream -> CompletableFuture.supplyAsync(() -> {
        ...
     }))
    .distinct()
    .toList();
CompletableFuture.allOf(prepareTablesTasks).join();
reduceExceptions(prepareTablesTasks, ...);
```